engine/src/main/java/org/apache/apex/engine/events/grouping/GroupingManager.java

```diff
@@ -153,14 +157,14 @@ public boolean removeOperatorFromGroupingRequest(int operatorId)
    */
   public void removeProcessedGroupingRequests()
   {
-    for (Entry<String, GroupingRequest> request : groupingRequests.entrySet()) {
-      if (request.getValue().getOperatorsToDeploy().size() == 0
-          && request.getValue().getOperatorsToUndeploy().size() == 0) {
-        LOG.info("Removing for :" + request.getKey());
-        groupingRequests.remove(request.getKey());
+    Iterator<Entry<String, GroupingRequest>> itr = groupingRequests.entrySet().iterator();
+    while (itr.hasNext()) {
+      Entry<String, GroupingRequest> request = itr.next();
+      if (request.getValue().isProcessed()) {
+        LOG.debug("Removing Grouping request for :" + request.getKey());
```

Comment:
Use {} parameters instead of string concat. It is a basic thing that all contributors should be aware of.